bin/strap.sh

```diff
@@ -254,156 +308,141 @@ fi
 if ! git config push.default >/dev/null; then
   git config --global push.default simple
 fi
-logk
 
 # Setup GitHub HTTPS credentials.
-logn "Setting up GitHub HTTPS credentials..."
-if git credential-osxkeychain 2>&1 | grep $Q "git.credential-osxkeychain" ; then
-  if [ "$(git config --global credential.helper)" != "osxkeychain" ]; then
-    logdebug "setting git credential helper to osxkeychain"
+if git credential-osxkeychain 2>&1 | grep $Q "git.credential-osxkeychain"; then
+  # Actually execute the credential in case it's a wrapper script for credential-osxkeychain
+  if git "credential-$(git config --global credential.helper 2>/dev/null)" 2>&1 \
+    | grep -v $Q "git.credential-osxkeychain"; then
     git config --global credential.helper osxkeychain
   fi
 
   if [ -n "$STRAP_GITHUB_USER" ] && [ -n "$STRAP_GITHUB_TOKEN" ]; then
-    logdebug "storing https credentials for github"
-    printf "protocol=https\nhost=github.com\n" | git credential-osxkeychain erase
-    printf "protocol=https\nhost=github.com\nusername=%s\npassword=%s\n" \
-          "$STRAP_GITHUB_USER" "$STRAP_GITHUB_TOKEN" \
-          | git credential-osxkeychain store
+    printf 'protocol=https\nhost=github.com\n' | git credential reject
+    printf 'protocol=https\nhost=github.com\nusername=%s\npassword=%s\n' \
+      "$STRAP_GITHUB_USER" "$STRAP_GITHUB_TOKEN" \
+      | git credential approve
   fi
 fi
 logk
 
-# add ssh key to github
```

Comment:
Assuming we continue with ssh-based authentication, we still need this.  (Though, I think the default/preferred key format has changed.)

Reply:
I get that.  But if we put this back, we still support automating ssh.  Otherwise, it's more for the wiki, right?

Reply:
this is in PR 40